mantidproject/mantid
# Make Load/SaveAscii2 aware of Distribution

**Description of work.**
- Set Distribution flag in file header for Load/SaveAscii
- Add Distribution property to LoadAscii to allow loading files without new flag in header
- Updated LoadAscii2 and SaveAscii2 tests
- Worth noting that if the LoadAscii2 parameter ColumnHeader is set to False(not default) but the workspace is Distribution=true, ignore the first and write a ColumnHeader anyway.

**To test:**

- Open Workbench
- Run CreateSampleWorkspace ...
1)
- Click the triangle by the workspace and notice that `Distribution=False`
- Select and run Save Ascii (maybe from the Workspace toolbox dropdown)
- Run Load on the new file and check it is also `Distribution=False`

2)
- Click the triangle by the workspace and notice that `Distribution=False`
- Select and run Save Ascii BUT make sure to set ColumnHeader=False
- Open the saved file in a text editor and check there is no commented column header
- Run Load on the new file and check it is also `Distribution=False`

3)
- Run ConvertToDistribution() on the workspace
- Click the triangle by the workspace and notice that `Distribution=True`
- Select and run Save Ascii (maybe from the Workspace toolbox dropdown)
- Run Load on the new file and check it is also `Distribution=True`

4)
- Run ConvertToDistribution() on the workspace
- Click the triangle by the workspace and notice that `Distribution=True`
- Select and run Save Ascii BUT make sure to set ColumnHeader=False (this will be ignored)
- Open the saved file in a text editor and check there IS a commented column header
- Run Load on the new file and check it is also `Distribution=True`

5)
- Click the triangle by the workspace and check that `Distribution=False`
- Select and run Save Ascii
- Run LoadAscii and set `ForceDistributionTrue=true` on the new file and check it is also `Distribution=True`

Fixes #34447 

Not added release notes or decided on PR base




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
